WordPress/openverse-frontend
# VLogoLoader

## Fixes


Fixes #371 by @zackkrida 

## Description



Adds a new `VLogoLoader` component and some supporting composables:

- One for using event listeners (no longer being used, so I could remove)
- One for using media queries (added by @obulat modified by me)
- One for using `prefers-reduced-motion`

The loader itself is an `aria-hidden`, visual enhancement to the site. It does **not** have any inherent accessibility qualities, nor is it focusable. Originally I thought this component should be an ARIA Live Region[^1] but after some research that will be a role better served by our actual search grid, which can announce it's own loading status, the # of newly added results after the user hits the 'load more' button, and so on. I'll make a new ticket soon to spec out those improvements. 

## Testing Instructions


This component can be tested with `npm run storybook` and viewing the `VLogoLoader` story. There are a few things you can do:

- Toggle the `status` prop from "loading" to "idle" with the handles
- Set `prefers-reduced-motion` in your browser while in the "loading" status, observe the change in animation to a more subtle, non-moving effect. This is currently a gentle pulse from full to partial opacity. @panchovm might have a better idea for a subtle animation that doesn't add movement, but still visually conveys to the user that loading is occurring. Prefers reduced motion !== wants _no_ motion. Here's a quick video on how to test for prefers reduced motion:

https://user-images.githubusercontent.com/6351754/144527413-bc1c3bd7-6d94-408e-8088-f283d9089e8d.mp4

## Follow-up questions

1. Please suggest any test cases I should add. I admittedly don't love unit testing of front-end components and am always looking for improvements there.
2. The component is extremely small in terms of pixel values in the mockups (cc @panchovm), with the logo being 20px tall with 14px of padding surrounding. This has been increased to 32px tall with 16px of padding in my component. Otherwise the Openverse logo is barely legible, even on my 5k monitor.
  
    > My sizing
    <img width="246" alt="CleanShot 2021-12-02 at 19 55 00@2x" src="https://user-images.githubusercontent.com/6351754/144527587-c20b8fa1-f3b7-4661-ba27-b5ee5d53693d.png">
  
    > Figma sizing
    <img width="120" alt="CleanShot 2021-12-02 at 19 54 04@2x" src="https://user-images.githubusercontent.com/6351754/144527622-8546808a-508e-4a17-852f-5421031741c4.png">
    <img width="165" alt="CleanShot 2021-12-02 at 19 54 29@2x" src="https://user-images.githubusercontent.com/6351754/144527635-a0246843-5803-43eb-9150-807496b17515.png">

3. The mobile menu includes the logotext "Openverse", I think that can simply exist _next to_ this loader but perhaps be wrapped in the same `<NuxtLink />` component.

## Checklist



[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

[^1]: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions
